xbmc/addons/AddonManager.cpp

```diff
@@ -397,10 +397,10 @@ std::vector<std::string> CAddonMgr::MigrateAddons()
   return DisableIncompatibleAddons(incompatible);
 }
 
-std::vector<std::string> CAddonMgr::DisableIncompatibleAddons(
+std::vector<AddonInfoPtr>& CAddonMgr::DisableIncompatibleAddons(
     const std::vector<AddonInfoPtr>& incompatible)
 {
-  std::vector<std::string> changed;
+  static std::vector<AddonInfoPtr> changed;
```

Comment:
Because the vector is only created here and is allocated on the heap (causing issues when the function goes out of scope). Since we need the return value it gets allocated on the stack and the callee can reference its contents. The alternative would be to pass the reference to the vector to be filled from CApp -> migrate add-ons -> disableaddons as an (in,out) argument. I didn't like that option as it would dissuade the method signature. Do you see any problem with this approach or may I leave it as it is?

Reply:
It should look like this:

```cpp
std::vector<AddonInfoPtr> CAddonMgr::DisableIncompatibleAddons(
    const std::vector<AddonInfoPtr>& incompatible)	    
{
  static std::vector<AddonInfoPtr> changed;
```

Returning by value and not reference.